server/handler.go

```diff
@@ -109,7 +109,12 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 }
 
 func (h *Handler) handleCert(w http.ResponseWriter, req *http.Request) {
-	ac := h.CertManager.GetAugmentedChain()
+	digest := strings.TrimPrefix(req.URL.Path, h.CertPath+"/")
+	ac, err := h.CertManager.Cache.Read(digest)
+	if err != nil {
+		replyServerError(w, xerrors.Errorf("serializing cert-chain: %w", err))
+		return
+	}
 
 	// TODO(yuizumi): Keep the previous AugmentedChains for a while; serve
```

Comment:
This if statement can be removed since it should be a contract of `Read` (even though it's currently not true).

Reply:
Sorry, I don't understand what you mean by this comment. Could you explain? Thanks.

Reply:
`Read` is expected to return an AugementedChain `ac` such that `ac.Digest` is equal to `digest`.[^1] Considering how `digest` is computed, `req.URL.Path` is equal to `path.Join(h.CertPath, digest)`, thus `path.Join(h.CertPath, ac.Digest)`. So the following if statement is meaningless.

[^1]: The current `DiskCache` does not behave that way, but it's rather a violation to the contract (though it's a temporary measure).

Reply:
By "if statement", I'm assuming you mean:

	if req.URL.Path != path.Join(h.CertPath, ac.Digest) 

Removed this.